lib/widgets/components/voucher/redeem_voucher_text_field.dart

```diff
@@ -0,0 +1,79 @@
+import 'package:coffeecard/base/strings.dart';
+import 'package:coffeecard/base/style/text_styles.dart';
+import 'package:coffeecard/cubits/voucher/voucher_cubit.dart';
+import 'package:coffeecard/data/repositories/v1/voucher_repository.dart';
+import 'package:coffeecard/service_locator.dart';
+import 'package:coffeecard/widgets/components/dialog.dart';
+import 'package:coffeecard/widgets/components/forms/text_field.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class RedeemVoucherTextField extends StatefulWidget {
+  const RedeemVoucherTextField();
+  @override
+  State<RedeemVoucherTextField> createState() => _RedeemVoucherTextFieldState();
+}
+
+class _RedeemVoucherTextFieldState extends State<RedeemVoucherTextField> {
+  final _controller = TextEditingController();
+  bool _loading = false;
+
+  String? _error;
+  String? get error => _error;
+  set error(String? error) => setState(() => _error = error);
+
+  void submit(BuildContext context) {
+    final text = _controller.text.trim();
+    if (text.isNotEmpty) context.read<VoucherCubit>().redeemVoucher(text);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider(
+      create: (_) => VoucherCubit(sl.get<VoucherRepository>()),
+      child: BlocConsumer<VoucherCubit, VoucherState>(
+        listener: (context, state) async {
```

Comment:
Extract listener to a new method. It could look like:
```dart
_listener(BuildContext context, VoucherState state) { ... }
```
Then this line would look like:
```dart
listener: _listener,
```